### DISCUSSION TITLE
Markup methods for Desktop menus in .NET MAUI

### DISCUSSION THREAD

**Discussion by egvijayanand** [CommunityToolkit/Maui.Markup#84] [New Feature Discussions]
Initiating this discussion for formal approval to induct the extension methods (Markups) that define the .NET MAUI multi-level menu structure (desktop form factor) completly in C#.

Code Sample:

```cs
MenuBarItems.Add(new MenuBarItem().Title("Locations")
                                 .AddMenuGroup(new MenuFlyoutSubItem().Title("Change Location")
                                                                      .AddSubMenuItem(new MenuFlyoutItem().Title("New York, USA"))
                                                                      .AddSubMenuItem(new MenuFlyoutItem().Title("London, UK"))
                                                                      .AddSubMenuItem(new MenuFlyoutItem().Title("Cape Town, RSA"))
                                                                      .AddSubMenuItem(new MenuFlyoutItem().Title("Beijing, PRC"))
                                                                      .AddSubMenuGroup(new MenuFlyoutSubItem().Title("India")
                                                                                                              .AddSubMenuItem(new MenuFlyoutItem().Title("Chennai, TN"))
                                                                                                              .AddSubMenuItem(new MenuFlyoutItem().Title("Kolkata, WB"))
                                                                                                              .AddSubMenuItem(new MenuFlyoutItem().Title("Mumbai, MH"))
                                                                                                              .AddSubMenuItem(new MenuFlyoutItem().Title("New Delhi, NCR"))))
                                 .AddMenuItem(new MenuFlyoutItem().Title("Add a Location").BindCommand("AddLocationCommand")));
```

Stakeholders can provide their valuable inputs for including the same into the markup toolkit.

Source files that represent this feature.

[MenuGroupExtensions.cs](https://github.com/egvijayanand/dotnet-maui-toolkit/blob/main/src/VijayAnand.MauiToolkit/VijayAnand.MauiToolkit/Markup/MenuGroupExtensions.cs)

[MenuItemExtensions.cs](https://github.com/egvijayanand/dotnet-maui-toolkit/blob/main/src/VijayAnand.MauiToolkit/VijayAnand.MauiToolkit/Markup/MenuItemExtensions.cs)

Notify to @brminnick @pictos @jfversluis

**Comment by TheCodeTraveler** [comment]
Thanks @egvijayanand! Love it!

The only question have one question: Does .NET MAUI have an interface we could reference? Ie instead of `where TMenuItem : BaseMenuItem`, is there an equivalent interface like `where TMenuItem : IMenuItem`?

**Reply by egvijayanand** [reply]
Need to check, that would make it align closer with MAUI.

**Reply by TheCodeTraveler** [reply]
Thanks! Totally, and it’ll match how we’ve implemented some of the other extension methods too 💯 

Here’s an example from `.Aspect()`: https://github.com/CommunityToolkit/Maui.Markup/blob/386e4c6a460b90ba5f8701a8efca6a53ce903676/src/CommunityToolkit.Maui.Markup/ImageExtensions.cs#L28

**Comment by CliffAgius** [comment]
Closed and answered
